eng/README.md

```diff
@@ -0,0 +1,36 @@
+# `Eng` directory
+
+The `Eng` directory contains source code for automated tooling running on this repository pull requests.
+
+For context on this directory, see [Design guidelines for spec repos validation tooling] (Microsoft-internal).
+
+## Code conventions
+
+Below are code convention we strive to follow in `eng` directory:
+
+### package.json
+
+- We align `package.json` dependencies versions across all `package.json` files.
+- We align `package.json` dependencies numbers with [microsoft/typespec package.json].
+  In few cases we allow more frequent update cadence.
+- We avoid doing package overrides. For example, we use `v8` of `eslint` instead of `v9` to avoid an override.
+  [See this comment for details][eslint override].
+- We order `package.json` keys as follows: `name private type main bin scripts engines dependencies devDependencies`.
```

Comment:
FYI this is my proposal for the key order. I moved `bin` and `scripts` close to top as package entry-points people probably want to refer to.